test/components/preprocessors/test_recursive_splitter.py

```diff
@@ -0,0 +1,258 @@
+import pytest
+
+from haystack import Document, Pipeline
+from haystack.components.preprocessors.recursive_splitter import RecursiveDocumentSplitter
+from haystack.components.preprocessors.sentence_tokenizer import SentenceSplitter
+
+
+def test_get_custom_sentence_tokenizer_success():
+    tokenizer = RecursiveDocumentSplitter._get_custom_sentence_tokenizer()
+    assert isinstance(tokenizer, SentenceSplitter)
+
+
+def test_init_with_negative_overlap():
+    with pytest.raises(ValueError):
+        _ = RecursiveDocumentSplitter(split_length=20, split_overlap=-1, separators=["."])
+
+
+def test_init_with_overlap_greater_than_chunk_size():
+    with pytest.raises(ValueError):
+        _ = RecursiveDocumentSplitter(split_length=10, split_overlap=15, separators=["."])
+
+
+def test_init_with_invalid_separators():
+    with pytest.raises(ValueError):
+        _ = RecursiveDocumentSplitter(separators=[".", 2])
+
+
+def test_init_with_negative_split_length():
+    with pytest.raises(ValueError):
+        _ = RecursiveDocumentSplitter(split_length=-1, separators=["."])
+
+
+def test_apply_overlap_no_overlap():
+    # Test the case where there is no overlap between chunks
+    splitter = RecursiveDocumentSplitter(split_length=20, split_overlap=0, separators=["."])
+    chunks = ["chunk1", "chunk2", "chunk3"]
+    result = splitter._apply_overlap(chunks)
+    assert result == ["chunk1", "chunk2", "chunk3"]
+
+
+def test_apply_overlap_with_overlap():
```

Comment:
Let's also add a test where we run this example with `split_length=20` and `split_overlap=6`. We want to catch the edge case where the second chunk doesn't fully capture the first chunk. See this PR https://github.com/deepset-ai/haystack/pull/8539 for how this happened with the NLTKDocumentSplitter.